ui/lib/pki/addon/components/pki-config/import.ts

```diff
@@ -0,0 +1,62 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+// Types
+import { HTMLElementEvent } from 'forms';
+import PkiConfigImportModel from 'vault/models/pki/config/import';
+import Store from '@ember-data/store';
+import Router from '@ember/routing/router';
+import FlashMessageService from 'vault/services/flash-messages';
+import errorMessage from 'vault/utils/error-message';
+
+interface File {
+  value: string;
+  fileName?: string;
+  enterAsText: boolean;
+}
+
+/**
+ * Pki Config Import Component creates a PKI Config record on mount, and cleans it up if dirty on unmount
+ */
+export default class PkiConfigImportComponent extends Component<Record<string, never>> {
+  @service declare readonly store: Store;
+  @service declare readonly router: Router;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked configModel: PkiConfigImportModel | null = null;
+
+  constructor(owner: unknown, args: Record<string, never>) {
+    super(owner, args);
+    const model = this.store.createRecord('pki/config/import');
+    this.configModel = model;
+  }
+
+  willDestroy() {
```

Comment:
Curious why you opted for `willDestroy` here as I haven't seen much use of the lifecycle hooks in glimmer components, aside from the constructor 💭 

The pattern I've seen us follow lately is a cancel action that unloads the record and then, if applicable, call the `this.args.onCancel` passed to the component which is the transition.

Reply:
unloading with `onCancel` is fine for when the user explicitly leaves the page, but in a case where, for example, the user clicks "back" or on another link, we also want the model to reset which is where the `willDestroy` hook comes in handy.

Reply:
Oh, one other reason -- since this form is being rendered inline on the page, we want the import config model to reset if the user clicks on a different option, so there aren't multiple config models floating around

Reply:
Ohhh, yes! That makes sense!